src/cpp/cpp_convert_type.cpp

```diff
@@ -19,125 +19,57 @@ Author: Daniel Kroening, kroening@cs.cmu.edu
 #include <util/invariant.h>
 #include <util/std_types.h>
 
+#include <ansi-c/ansi_c_convert_type.h>
 #include <ansi-c/gcc_types.h>
 
 #include "cpp_declaration.h"
 #include "cpp_name.h"
 
-class cpp_convert_typet
+class cpp_convert_typet : public ansi_c_convert_typet
 {
 public:
-  unsigned unsigned_cnt, signed_cnt, char_cnt, int_cnt, short_cnt,
-           long_cnt, const_cnt, restrict_cnt, constexpr_cnt, volatile_cnt,
-           double_cnt, float_cnt, complex_cnt, cpp_bool_cnt, proper_bool_cnt,
-           extern_cnt, noreturn_cnt, wchar_t_cnt, char16_t_cnt, char32_t_cnt,
-           int8_cnt, int16_cnt, int32_cnt, int64_cnt, ptr32_cnt, ptr64_cnt,
-           float80_cnt, float128_cnt, int128_cnt;
+  std::size_t wchar_t_count, char16_t_count, char32_t_count;
```

Comment:
It may be worth adding a comment here that these do exist in C11, but are typedefs, whereas they are keywords in C++ since in C++11. This is in essence the reason why the C++ converter differs from the C one.

Reply:
Good point - I've added a comment, echoing pretty much what you said.